keps/sig-auth/541-external-credential-providers/README.md

```diff
@@ -184,11 +244,36 @@ Credentials may become invalid (e.g. expire) after being returned by the
 provider but before `expirationTimestamp` in the returned `ExecCredential`.
 
 Credential provider should ensure validity of the credentials it returns and
-return an error if it can't provide valid credentials.
+return an error if it cannot provide valid credentials.
+
+In case client gets `401 Unauthorized` response status from remote endpoint when
+using credentials from a provider, client should re-execute the provider,
+ignoring `expirationTimestamp`.
+
+### Test Plan
 
-In case client gets `401 Unauthorized` or `403 Forbidden` response status from
-remote endpoint when using credentials from a provider, client should
-re-execute the provider, ignoring `expirationTimestamp`.
+Unit tests to confirm:
+
+- Version mismatch is detected
+- Credentials are cached in-memory correctly
+  + Executable is only called as needed
+  + Expired credentials are rotated automatically
+  + Credentials are used across many requests (as long as they are still valid)
+- Single flight all calls to a given executable (when the config is the same)
+- Reasonable timeout to executable calls so clients do not hang indefinitely
+
+Integration (or e2e CLI) tests to confirm:
+
+- Shared informers backed by exec credential work as expected
+  + Credential rotation does not cause issues
+  + Transient failures are correctly retried
+  + Executables requiring interactive prompts fail gracefully
+  + Executables are not called in a hot loop during transient failure
+- Static forms of auth should interact correctly with exec credential plugin
+  + Basic auth
+  + Token based auth
+  + Cert based auth
+- Interactive login flows work
```

Comment:
Sub test of this is coverage tty forwarding.

Reply:
Done.